examples/chef/common/chef-rvc-operational-state-delegate.cpp

```diff
@@ -46,100 +60,115 @@ CHIP_ERROR GenericOperationalStateDelegateImpl::GetOperationalPhaseAtIndex(size_
     return CopyCharSpanToMutableCharSpan(mOperationalPhaseList[index], operationalPhase);
 }
 
-void GenericOperationalStateDelegateImpl::HandlePauseStateCallback(GenericOperationalError & err)
+void RvcOperationalStateDelegate::HandlePauseStateCallback(GenericOperationalError & err)
 {
     // placeholder implementation
     auto error = GetInstance()->SetOperationalState(to_underlying(OperationalState::OperationalStateEnum::kPaused));
     if (error == CHIP_NO_ERROR)
     {
-        err.Set(to_underlying(ErrorStateEnum::kNoError));
+        err.Set(to_underlying(OperationalState::ErrorStateEnum::kNoError));
     }
     else
     {
-        err.Set(to_underlying(ErrorStateEnum::kUnableToCompleteOperation));
+        err.Set(to_underlying(OperationalState::ErrorStateEnum::kUnableToCompleteOperation));
     }
 }
 
-void GenericOperationalStateDelegateImpl::HandleResumeStateCallback(GenericOperationalError & err)
+void RvcOperationalStateDelegate::HandleResumeStateCallback(GenericOperationalError & err)
 {
     // placeholder implementation
-    auto error = GetInstance()->SetOperationalState(to_underlying(OperationalStateEnum::kRunning));
+    auto error = GetInstance()->SetOperationalState(to_underlying(OperationalState::OperationalStateEnum::kRunning));
     if (error == CHIP_NO_ERROR)
     {
-        err.Set(to_underlying(ErrorStateEnum::kNoError));
+        err.Set(to_underlying(OperationalState::ErrorStateEnum::kNoError));
     }
     else
     {
-        err.Set(to_underlying(ErrorStateEnum::kUnableToCompleteOperation));
+        err.Set(to_underlying(OperationalState::ErrorStateEnum::kUnableToCompleteOperation));
     }
 }
 
-void GenericOperationalStateDelegateImpl::HandleStartStateCallback(GenericOperationalError & err)
+void RvcOperationalStateDelegate::HandleStartStateCallback(GenericOperationalError & err)
 {
+    OperationalState::GenericOperationalError current_err(to_underlying(OperationalState::ErrorStateEnum::kNoError));
+    GetInstance()->GetCurrentOperationalError(current_err);
+
+    if (current_err.errorStateID != to_underlying(OperationalState::ErrorStateEnum::kNoError))
+    {
+        err.Set(to_underlying(OperationalState::ErrorStateEnum::kUnableToStartOrResume));
+        return;
+    }
```

Comment:
Why do we override this to UnableToStartOrResume here, and not in cluster logic?

Reply:
I think this is respecting the spec of 1.15.6.3 Start Command 
```
A device that is unable to honor the Start command for whatever reason SHALL respond 
with an "OperationalCommandResponse" command 
with an ErrorStateID of UnableToStartOrResume 
but take no further action.
```
![image](https://github.com/project-chip/connectedhomeip/assets/7869096/b5d9a3d3-6943-4290-8707-80bbe9333b5d)